lib/Parse/ParseDecl.cpp

```diff
@@ -970,7 +972,20 @@ bool Parser::parseDifferentiableAttributeArguments(
   SyntaxParsingContext ContentContext(
       SyntaxContext, SyntaxKind::DifferentiableAttributeArguments);
 
-  // Parse optional differentiation parameters, starting with the 'wrt:' label.
+  // Parse optional differentiation parameters.
+  // Parse 'linear' label (optional).
+  if (Tok.is(tok::identifier) && Tok.getText() == "linear") {
+    linear = true;
+    consumeIdentifier();
+    // If no trailing comma or 'where' clause, terminate parsing arguments.
+    if (Tok.isNot(tok::comma) && Tok.isNot(tok::kw_where))
+      return false;
+    if (consumeIfTrailingComma())
+      return errorAndSkipToEnd();
+  } else {
```

Comment:
I'd suggest removing the `else` case and initialize the definition of `linear` with `false`.